### PR TITLE
Add type definitions for unused-webpack-plugin

### PR DESCRIPTION
**Pull request by remcohaszing**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/unused-webpack-plugin/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { WebpackPluginInstance, Compiler } from 'webpack';
+
+export = UnusedWebpackPlugin;
+
+declare namespace UnusedWebpackPlugin {
+    interface UnusedWebpackPluginOptions {
+        /**
+         * array of directories where to look for unused source files.
+         */
+        directories: string[];
```
Comment (martin-badin) [inline types/unused-webpack-plugin/index.d.ts:16]: this field is optional and the default value is an empty array
https://github.com/MatthieuLemoine/unused-webpack-plugin/blob/master/index.js#L6

File: types/unused-webpack-plugin/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        /**
+         * array of exclude patterns when looking for unused source files.
+         */
+        exclude?: string[];
+
+        /**
+         * Root directory that will be use to display relative paths instead of absolute ones
+         */
+        root?: string;
```
Comment (martin-badin) [inline types/unused-webpack-plugin/index.d.ts:26]: this field is required 
https://github.com/MatthieuLemoine/unused-webpack-plugin/blob/master/index.js#L8

File: types/unused-webpack-plugin/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    interface UnusedWebpackPluginOptions {
+        /**
+         * array of directories where to look for unused source files.
+         */
+        directories: string[];
+
+        /**
+         * array of exclude patterns when looking for unused source files.
+         */
+        exclude?: string[];
```
Comment (martin-badin) [inline types/unused-webpack-plugin/index.d.ts:21]: the default value is an empty array
https://github.com/MatthieuLemoine/unused-webpack-plugin/blob/master/index.js#L7

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/unused-webpack-plugin/index.d.ts (hunk 3 above)
Comment (remcohaszing) [reply types/unused-webpack-plugin/index.d.ts:25]: You’re right. I feel like most DefinitelyTyped reviewers don’t pay any attention to the docs. I appreciate that you do so. :smile:

File: types/unused-webpack-plugin/index.d.ts (hunk 2 above)
Comment (remcohaszing) [reply types/unused-webpack-plugin/index.d.ts:30]: Although the constructor doesn’t assign a default value, the readme explicitly shows that the root is optional https://github.com/MatthieuLemoine/unused-webpack-plugin#options

Comment (martin-badin) [reply types/unused-webpack-plugin/index.d.ts:30]: good. so all options are optional

**Review by martin-badin — APPROVED** [review]

File: types/unused-webpack-plugin/index.d.ts (hunk 3 above)
Comment (DanielRosenwasser) [reply types/unused-webpack-plugin/index.d.ts:25]: Agreed, thank you for helping out @martin-badin!

**Pull request merged**